x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/oidc/OpenIdConnectPrepareAuthenticationRequest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.security.action.oidc;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+
+/**
+ * Represents a request to prepare an OAuth 2.0 authentication request
```

Comment:
The naming convention we use in this case is a little confusing. A `OpenIdConnectPrepareAuthenticationRequest` is a request that Kibana would make to Elasticsearch in order for Elasticsearch to create an `OpenID Connect Authentication Request` (which is basically an `oAuth 2.0 Authorization Request`). Elasticsearch replies to Kibana with an `OpenIdConnectPrepareAuthenticationResponse` which contains this `oAuth 2.0 Authorization Request` so that Kibana can redirect the use to the OP.